### PR TITLE
Add user confirmation for .ycm_extra_conf loading

### PR DESCRIPTION
**Pull request by aarentoft**
Here is an attempt at fixing https://github.com/LuckyGeck/YcmdCompletion/issues/8. The implementation is based on the discussion [here](https://github.com/abingham/emacs-ycmd/issues/19) and the implementation of .ycm_extra_conf handling in vim YCM.

### CODE REVIEW

File: ycmd/http_client.py [hunk 1]
```diff
@@ -128,8 +135,15 @@ def _CallHttp(self, method, handler, data=None):
...
             HMAC_HEADER, self._HmacForRequest(method, handler, data))
         req.data = bytes(data, 'utf-8')
-        readData = urlopen(req).read().decode('utf-8')
-        return readData
+        try:
+            readData = urlopen(req).read().decode('utf-8')
+            return readData
+        except HTTPError as err:
+            if err.code == 500:
+                responseAsJson = json.loads( err.read().decode('utf-8') )
```
Comment (LuckyGeck) [inline ycmd/http_client.py]: Please, stick to pep8 here - remove redundant spaces from json.loads(..) call.

**Comment by LuckyGeck** [comment]
It's great, that you have implemented this feature - thank you very much!
Diff LGTM
Have you tested this solution on real project with .ycm_extra_conf-s?

**Comment by aarentoft** [comment]
Yes, I have tested with two of my own C++ projects and so far YcmdCompletion has behaved as Vim and YouCompleteMe do when working with these projects.

**Comment by LuckyGeck** [comment]
Greate, thanks! Merged to master.

**Comment by trevordixon** [comment]
YcmdCompletion doesn't seem to have been updated in Package Manager since this was fixed.

**Pull request merged**